### PR TITLE
fix: [ANDROAPP-2984] Crash when using fingerprint

### PR DESCRIPTION
**Pull request by Balcan**
[issue](https://jira.dhis2.org/browse/ANDROAPP-2984)

### CODE REVIEW

File: app/src/main/java/org/dhis2/usescases/login/LoginActivity.kt [hunk 1]
```diff
@@ -414,18 +399,16 @@ class LoginActivity : ActivityGlobalAbstract(), LoginContracts.View {
...
     }
 
     override fun getDefaultServerProtocol(): String =
         getString(R.string.login_https)
+
+    override fun getPromptParams(): Goldfinger.PromptParams =
+        Goldfinger.PromptParams.Builder(this)
+            .title(R.string.fingerprint_title)
+            .negativeButtonText(R.string.cancel)
+            .build()
```
Comment (JaimeToca) [inline app/src/main/java/org/dhis2/usescases/login/LoginActivity.kt:413]: :100:

File: app/src/main/java/org/dhis2/data/fingerprint/FingerPrintControllerImpl.kt [hunk 2]
```diff
@@ -1,17 +1,23 @@
...
+import co.infinum.goldfinger.Goldfinger
 import co.infinum.goldfinger.rx.RxGoldfinger
 import io.reactivex.Observable
 
-class FingerPrintControllerImpl(val goldfinger: RxGoldfinger, val mapper: FingerPrintMapper) :
+class FingerPrintControllerImpl(
+    val context: Context,
+    val goldfinger: RxGoldfinger,
+    val mapper: FingerPrintMapper
+) :
```
Comment (JaimeToca) [inline app/src/main/java/org/dhis2/data/fingerprint/FingerPrintControllerImpl.kt:12]: I think we don't need the context :+1:

File: app/src/test/java/org/dhis2/usescases/login/LoginPresenterTest.kt [hunk 3]
```diff
@@ -239,13 +239,12 @@ class LoginPresenterTest {
     @Test
     fun `Should display message and hide fingerprint dialog when authenticate throws an error`() {
         whenever(
```
Comment (JaimeToca) [inline app/src/test/java/org/dhis2/usescases/login/LoginPresenterTest.kt:241]: should we change the test name? Is not hidingFingerPrintDialog anymore?

Comment (Balcan) [reply app/src/test/java/org/dhis2/usescases/login/LoginPresenterTest.kt]: Done!

File: app/src/main/java/org/dhis2/data/fingerprint/FingerPrintControllerImpl.kt (hunk 2 above)
Comment (Balcan) [reply app/src/main/java/org/dhis2/data/fingerprint/FingerPrintControllerImpl.kt]: Done!

**Comment by JaimeToca** [comment]
@Balcan moving to ready for testing. Reviewers were deleted after push

**Pull request merged**